### ISSUE TITLE
DualModeConnect_IPAddressListToHost_Throws failed in CI

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#17362]
https://ci.dot.net/job/dotnet_corefx/job/master/job/centos7.1_debug_prtest/4847/consoleText

```     System.Net.Sockets.Tests.DualModeConnectToIPAddressArray.DualModeConnect_IPAddressListToHost_Throws(connectTo: [127.0.0.1], listenOn: ::, dualModeServer: False) [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.Net.Sockets.SocketException)
        Actual:   (No exception was thrown)
        Stack Trace:
           /mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/src/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs(283,0): at System.Net.Sockets.Tests.DualModeConnectToIPAddressArray.DualModeConnect_IPAddressListToHost_Throws(IPAddress[] connectTo, IPAddress listenOn, Boolean dualModeServer)
```

**Comment by karelz** [comment]
cc @steveharter @Priya91

**Comment by mellinoe** [comment]
Hit again: https://ci.dot.net/job/dotnet_corefx/job/master/job/centos7.1_release_prtest/6108/consoleText

**Comment by steveharter** [comment]
At least five failures from 3/7-4/7
```
dotnet_corefx/master/centos7.1_debug_prtest 4215 	centos71-20170216-d1bce0 	03/07 10:35 AM 	qmfrederik 	16794
dotnet_corefx/master/ubuntu14.04_debug 1493 	ubuntu1404-20160211-1-f5c721 	03/13 09:17 PM 		0
dotnet_corefx/master/opensuse13.2_release 1383 	suse132-20160315-7ba110 	03/14 03:37 PM 		0
dotnet_corefx/master/outerloop_netcoreapp_ubuntu16.04_debug 8 	ubuntu1604-20170216-outere8ee30 	04/07 01:30 PM 		0
```
+ Mellinoe's on 4/12 which is not yet in jdash.

**Comment by karelz** [comment]
Also fixed by #18576
